subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractMutableModuleComponentResolveMetadata.java

```diff
@@ -272,12 +280,17 @@ public void setArtifactOverrides(Iterable<? extends ModuleComponentArtifactMetad
 
     @Override
     public List<? extends DependencyMetadata> getDependencies() {
```

Comment:
:x: This should be a protected method called `getImmutableDependencies()`, used by `asImmutable()`.

Reply:
That sounds good. Changed it.